plugins/resource_management/app/controllers/resource_management/domain_admin_controller.rb

```diff
@@ -68,6 +70,55 @@ def update
       end
     end
 
+    def reduce_quota
+      # Note: this function is used in two ways
+      # first show the reduce quota window
+      # second: reduce the quota
+      
+      prepare_data_for_details_view(@resource.service.to_sym, @resource.name.to_sym)
+      unless params[:resource]
+        # show the reduce quota window 
+      else
+        # reduce the quota
+        value = params[:resource][:approved_quota]
+        
+        if value.empty?
+          @resource.add_validation_error(:approved_quota, "empty value is invalid")
+        else 
+          begin
+            # pre check value
+            if @resource.approved_quota < @resource.data_type.parse(value) 
+                @resource.add_validation_error(:approved_quota, "wrong value: because the reduced quota value of #{value} is higher than your approved quota")
+            elsif @resource_status[:current_quota_sum] > @resource.data_type.parse(value)
+              @resource.add_validation_error(:approved_quota, "wrong value: it is now allowed to reduce the quota below your current used quota value of #{@resource_status[:current_quota_sum]}")
+            elsif @resource.approved_quota == @resource.data_type.parse(value) 
+                @resource.add_validation_error(:approved_quota, "wrong value: because the reduced quota value is the same as your approved quota")
+            else
+              @resource.approved_quota = @resource.data_type.parse(value)
+            end
+          rescue ArgumentError => e
+            @resource.add_validation_error(:approved_quota, 'is invalid: ' + e.message)
+          end
+        end
+        
+        # save the new quota to database
+        if @resource.save
+          @area = @resource.service.to_sym
+          @area_services = ResourceManagement::ServiceConfig.in_area(@area).map(&:name)
+          prepare_data_for_resource_list(@area_services)
```

Comment:
Nevermind. I was just confused because the `else` has an explicit `render action: 'reduce_quota'`, which is the same as what's happening here implicitly.